inotify.go

```diff
@@ -315,21 +297,7 @@ func (w *Watcher) readEvents() {
 // channel. Such as events marked ignore by the kernel, or MODIFY events
 // against files that do not exist.
 func (e *Event) ignoreLinux(mask uint32) bool {
-	// Ignore anything the inotify API says to ignore
-	if mask&unix.IN_IGNORED == unix.IN_IGNORED {
-		return true
-	}
-
-	// If the event is Create or Write, the file must exist, or the
-	// event will be suppressed.
-	// *Note*: this was put in place because it was seen that a Write
-	// event was sent after the Remove. This ignores the Write and
-	// assumes a Remove will come or has come if the file doesn't exist.
-	if e.Op&Create == Create || e.Op&Write == Write {
-		_, statErr := os.Lstat(e.Name)
-		return os.IsNotExist(statErr)
-	}
```

Comment:
I think the epoll replacement and removal of the lstat are two unrelated things?
Or are they related somehow?

I'm a bit nervous about removing it, since I assume it was added for a reason to solve a specific issue.  Looks like it was added in 2013:

https://github.com/fsnotify/fsnotify/commit/cc2c34eeb074ea38c16dbea8600ba03ea36bc9b0#diff-b5e2484a4976be237475e7872e566a59aac5d8cc82223bdba01dbd61c2d69558R243

Issue 36 refers to this issue on the old repo, which mentions it fixes a memory leak: https://github.com/howeyc/fsnotify/issues/36

I think it would be better to remove it here and create a new PR, so we can focus on just one thing, which is already a large change.

Reply:
They're unrelated, I just didn't see the purpose of using lstat here, since it provides no real guarantees, and figured I'd remove it too. I can move that change to a separate PR, np there.

Reply:
> I just didn't see the purpose of using lstat here, since it provides no real guarantees

I agree; I also don't see the purpose (and think the current behaviour is confusing), which is why I'm hesitant to "just remove it". Chesterton's fence and all of that.

Reply:
Opened #470 for this